<hr>
<br>
<center><b>EDUARDO MENDIETA<br><br>Unidad II: Actividad 2: LIMPIEZA DE DATOS_parte 1</b></center>
<br>
<hr>
<b>Importando los datos de un archivo excel en un dataframe de Pandas:</b>

In [12]:
import pandas as pd


df = pd.read_excel('datos.xlsx', sheet_name=0)
df.head(10)

Nombre  Edad  Puntuacion Sexo
0  Anthony Wallace  46.0        86.0    M
1    Steven Barker  25.0        67.0    F
2     Barry Graham  56.0        82.0    M
3   Austin Jackson  40.0        92.0    M
4  Michael Ramirez   NaN        63.0    F
5     Karen Ashley  53.0         NaN  NaN
6  Zachary Sampson  20.0        85.0    M
7     Karen Larson   NaN         NaN    M
8    Heather Moore  61.0        60.0    F
9       James Ross  19.0        96.0    M

<hr>
<b>Obteniendo información del dataframe:</b>

In [13]:
print(f'\nInformacion:\n\n{df.info()}')

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2000 entries, 0 to 1999
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Nombre      2000 non-null   object 
 1   Edad        1799 non-null   float64
 2   Puntuacion  1570 non-null   float64
 3   Sexo        1331 non-null   object 
dtypes: float64(2), object(2)
memory usage: 62.6+ KB

Informacion:

None


<hr>
<b>Identificando valores nulos por columna:</b>

In [14]:
nulos_por_columna = df.isnull().sum()
print(f'\nNulos por columna:\n\n{nulos_por_columna}')


Nulos por columna:

Nombre          0
Edad          201
Puntuacion    430
Sexo          669
dtype: int64


<hr>
<b>En la columna 'Edad' se identifican 201 registros nulos, por lo que se procede a reemplazar por el promedio de las edades:</b>

In [15]:
df['Edad'] = df['Edad'].fillna('0').astype(int)

promedio = round(df[df['Edad'] != 0]['Edad'].mean())
print(f'\nPromedio de edades: {promedio}\n')

df['Edad'] = df['Edad'].replace(0, promedio)

df.head(10)


Promedio de edades: 41



Nombre  Edad  Puntuacion Sexo
0  Anthony Wallace    46        86.0    M
1    Steven Barker    25        67.0    F
2     Barry Graham    56        82.0    M
3   Austin Jackson    40        92.0    M
4  Michael Ramirez    41        63.0    F
5     Karen Ashley    53         NaN  NaN
6  Zachary Sampson    20        85.0    M
7     Karen Larson    41         NaN    M
8    Heather Moore    61        60.0    F
9       James Ross    19        96.0    M

<hr>
<b>En la columna 'Puntuacion' se identifican 430 registros nulos, en este caso cambiaremos los valores nulos por el valor modal:</b

In [17]:
df['Puntuacion'] = df['Puntuacion'].fillna(-1).astype(int)

valor_modal = round(df[df['Puntuacion'] != -1]['Puntuacion'].mode()[0])
print(f'\nValor modal: {valor_modal}\n')

df['Puntuacion'] = df['Puntuacion'].replace(-1, valor_modal)

print(df.head(10))


Valor modal: 72

            Nombre  Edad  Puntuacion Sexo
0  Anthony Wallace    46          86    M
1    Steven Barker    25          67    F
2     Barry Graham    56          82    M
3   Austin Jackson    40          92    M
4  Michael Ramirez    41          63    F
5     Karen Ashley    53          72  NaN
6  Zachary Sampson    20          85    M
7     Karen Larson    41          72    M
8    Heather Moore    61          60    F
9       James Ross    19          96    M


<hr>
<b>En la columna 'Sexo' se identifican 669 registros nulos, en este caso cambiaremos los valores nulos por el valor 'Desconocido':</b

In [18]:
df['Sexo'] = df['Sexo'].fillna('Desconocido')
df.head(10)

Nombre  Edad  Puntuacion         Sexo
0  Anthony Wallace    46          86            M
1    Steven Barker    25          67            F
2     Barry Graham    56          82            M
3   Austin Jackson    40          92            M
4  Michael Ramirez    41          63            F
5     Karen Ashley    53          72  Desconocido
6  Zachary Sampson    20          85            M
7     Karen Larson    41          72            M
8    Heather Moore    61          60            F
9       James Ross    19          96            M

<hr>
<b>Exportamos los datos limpios en un nuevo archivo de Excel:</b>

In [19]:
df.to_excel('datos_limpios.xlsx', index=False)